src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -46,4 +47,29 @@ public void testShorexTotalHitsValidate() {
                 .isEqualTo("52")
                 .as("Inside Summary, Total hit is not 52");
     }
+
+    /**
+     * Checking if total number of shorex is 52 or not.
+     */
+    @Test
+    public void testShorexProductTypeValidate() {
+        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam)
+                .as(Response.class);
+
+        int counter = 0;
+
+        for (int i = 0; i < response.getPayload().getProducts().size(); i++) {
+            if (response.getPayload().getProducts().get(i).getProductType().getProductType()
+                    .equalsIgnoreCase("SHOREX")) {
+                counter = counter + 1;
+            }
+        }
+
+        Assertions.assertThat(response.getStatus()).isEqualTo(200)
+                .as("Json response status is not 200");
+
+        softAssert.assertThat(counter == 52)
+                .as("Total number of SHOREX is not 52")
+                .isTrue();
```

Comment:
do not use assert on the counter.
Instead use a soft assert inside the for loop